### ISSUE TITLE
Decrypt fail when using the GnuPG-Agent

### ISSUE THREAD

**Issue by llernestal** [TheLocehiliosan/yadm#20]
When I ran `yadm encrypt`, then `yadm decrypt` and having set the `yadm.gpg-recipient` to my personal gpg key I got a **Bad passphrase** error. Without being prompted for passphrase when decrypting.

I changed https://github.com/TheLocehiliosan/yadm/blob/master/yadm#L206 to:
```bash
(gpg --no-use-agent -d "$YADM_ARCHIVE" || echo 1) | tar v$tar_option -C "$YADM_WORK"
```
and then I was prompted for passphrase when decrypting. I'm on Ubuntu 16.04, gpg version 1.4.20

Maybe it's not really an issue. But I thought it might help somebody running in to the same problem.

**Comment by TheLocehiliosan** [comment]
I have been unable to replicate your error. Perhaps you could provide a step-by-step process to recreate the problem? What version of **yadm** are you using? For me `apt-get install yadm` installs version `1.03`, which does not yet support the `yadm.gpg-recipient` configuration.

I think, in general, the way to disable gpg-agent is to ensure that the environment variable `GPG_AGENT_INFO` is unset or empty. Without modifying **yadm** you could use an alias like this:

    alias yadm="GPG_AGENT_INFO= yadm"

I have a hunch that the problem you're describing has more to do with your gpg-agent configuration. You may be able to test your basic GPG usage like this:

    echo testing | gpg -e -r <recipient> | gpg

Or indeed, you could try decrypting the archive itself, and see the result:

    gpg -d <yadm archive file> | tar tvf -

**Comment by llernestal** [comment]
I use version `1.04` of **yadm**. Installed it using:

```
curl -fLo /usr/local/bin/yadm https://github.com/TheLocehiliosan/yadm/raw/master/yadm && chmod a+x /usr/local/bin/yadm
```

After running the tests you suggested I found out that the problem only occurred when running `gpg` inside a **tmux** session. After a number of googles [I found this Stackoverflow answer](http://stackoverflow.com/a/9833996/4116624 "How do I start tmux with my current environment?"). Then I added this line to my `tmux.conf`:

```
set-option -ga update-environment ' GPG_AGENT_INFO'
```

Now I get the passphrase prompt as expected inside **tmux** sessions. So, yes, you were correct. The problem was with my gpg-agent configuration. Thanks for pointing me in the right direction! :)
